src/Microsoft.IdentityModel.Protocols.WsFederation/WsFederationMetadataSerializer.cs

```diff
@@ -154,20 +176,40 @@ protected virtual void ReadSecurityTokenServiceTypeRoleDescriptor(WsFederationCo
             // Process <RoleDescriptorr>
             reader.ReadStartElement();
 
+            // Warn if there is no KeyDescriptor element for Signing
+            // Throw if there is no SecurityTokenEndpoint element
+            var hasKeyDescriptorForSigning = false;
+            var hasSecurityTokenEndpoint = false;
+
             while (reader.IsStartElement())
             {
                 if (IsKeyDescriptorForSigning(reader))
```

Comment:
Do the same check you do below.
if (reader.IsStartElement(WsFederationConstants.Elements.KeyDescriptor, ns)

We don't need that methog Is...